marconi-chain-index/src/Marconi/ChainIndex/Logging.hs

```diff
@@ -54,6 +59,14 @@ marconiFormatting =
   Pretty.renderStrict
     . Pretty.layoutPretty Pretty.defaultLayoutOptions
 
+logMInfo :: MonadIO m => Pretty.Doc () -> ChainIndexerT m ()
+logMInfo msg = do
+  trace <- Lens.view runIndexerConfigTrace
+  lift $ logInfo trace msg
+
+-- TODO: this should also throw an exception
+logMError = undefined
```

Comment:
I feel like log functions shouldn't be throwing exceptions, despite what we said earlier today about having a unified framework for both.

Perhaps it would be OK to have a function that does both, but have that reflected in the name. Something like `logThrowError`.

In another system I worked on, we also included a "context" parameter which was just a string that indicated where the error was thrown from. This helped a lot when looking at logs.

Reply:
I agree, I think we need to have the concrete requirements regarding errors and exceptions in order to make any changes here.